src/layout/actions/layout.rs

```diff
@@ -924,27 +924,32 @@ impl LayoutTree {
             let parent_ix = self.tree.parent_of(child_ix)
                 .expect("Node had no parent");
             let children = self.tree.children_of(parent_ix);
-            let index = children.iter().position(|&node_ix| node_ix == child_ix);
-            let index_str = index.map(|num| (num + 1).to_string());
 
             // I can't do it inside the match ahead because it borrows
             // self.tree, and I need self.tree to get the children's titles
-            let titles = children.iter()
-                .filter(|&&node_ix| {
-                    !self.tree[node_ix].floating()
-                })
-                .map(|&node_ix| {
+            let mut titles = Vec::<String>::with_capacity(children.len());
+            let mut index = None;
+            let mut filtered_idx = -1;
+            for &node_ix in children.iter() {
```

Comment:
I don't really like this iterative style. I find it much harder to read compared to the previous version. It might be more efficient due to not looping through the list twice, but the list will always be small and it will be easier to read that way.

One thing that will help: change children creation on line 926 to use `self.tree.grounded_children`, I'm not sure why I didn't switch to that in the first place to be honest. That will remove the need for the filter.